### PR TITLE
Ptrmu map class

### PR DESCRIPTION
**Pull request by ptrmu**
Slow progress. Nothing is working yet.

### CODE REVIEW

File: flock_vlam/include/map.hpp [hunk 1]
```diff
@@ -0,0 +1,120 @@
+#ifndef FLOCK_VLAM_MAP_H
+#define FLOCK_VLAM_MAP_H
+
+#include <map>
+#include <flock_vlam_msgs/msg/observations__struct.hpp>
```
Comment (clydemcqueen) [inline flock_vlam/include/map.hpp]: What is this? Doesn't this get included with msg/map.hpp?

File: flock_vlam/src/map.cpp [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+#include "opencv2/calib3d.hpp"
+
+namespace flock_vlam
+{
+
+  geometry_msgs::msg::PoseWithCovarianceStamped TransformWithCovariance::to_msg(
+    std_msgs::msg::Header & header)
+  {
+    geometry_msgs::msg::PoseWithCovarianceStamped msg;
+    msg.set__header(header);
```
Comment (clydemcqueen) [inline flock_vlam/src/map.cpp]: Interesting, I've never notice these. Why not use msg.header = header; ?

File: flock_vlam/include/map.hpp (hunk 1 above)
Comment (ptrmu) [reply flock_vlam/include/map.hpp]: Observations msg contains the marker 2d corners from an image. Map msg contains poses of markers in the map frame. Or that is the concept.

File: flock_vlam/src/map.cpp (hunk 2 above)
Comment (ptrmu) [reply flock_vlam/src/map.cpp]: I don't know. I thought I tried that. I will switch.

**Comment by ptrmu** [comment]
git doesn't create directories that don't have files in them. my cfg directory is empty at this point. You could just create the directory.

**Comment by clydemcqueen** [comment]
Got it, added cfg and launch dirs, builds nicely.

File: flock_vlam/include/map.hpp (hunk 1 above)
Comment (clydemcqueen) [reply flock_vlam/include/map.hpp]: Got it.

#include "flock_vlam_msgs/msg/observations.hpp" also works for me

Comment (ptrmu) [reply flock_vlam/include/map.hpp]: Good eye. I misinterpreted your question. I'm not sure where that came from. I have a vague feeling that I did something at some point where headers were auto-inserted. I will fix it.

**Pull request merged**